framework/src/modules/BaseInternalMethod.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright © 2022 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+import { StoreGetter, ImmutableStoreGetter } from './base_store';
+import { NamedRegistry } from './named_registry';
+
+export abstract class BaseInternalMethod {
+	public readonly events: NamedRegistry;
+	public readonly context: StoreGetter | ImmutableStoreGetter;
+	protected readonly stores: NamedRegistry;
+
+	public constructor(
+		stores: NamedRegistry,
+		events: NamedRegistry,
+		context: StoreGetter | ImmutableStoreGetter,
```

Comment:
it's better to pass the context per method than providing it at the class level similar to method.
type-wise, each method can specify if it's immutable or not

Reply:
Since this PR is getting heavy, I have only removed it from `BaseInternalMethod` we can tackle it in separate issue for removing it from `BaseInteroperabilityInternalMethod` as that will require another round of changes to each function and its context https://github.com/LiskHQ/lisk-sdk/issues/7702